sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -76,8 +76,19 @@ object Literal {
       val decimal = Decimal(d)
       Literal(decimal, DecimalType.fromDecimal(decimal))
     case d: JavaBigDecimal =>
-      val decimal = Decimal(d)
-      Literal(decimal, DecimalType.fromDecimal(decimal))
+      Literal(Decimal(d), DecimalType(Math.max(d.precision, d.scale), d.scale()))
+      if (d.abs().compareTo(new JavaBigDecimal("0")) == 0) {
```

Comment:
JavaBigDecimal.ZERO
But wait is this only a problem for Spark 2.4 + Hive 1.x? that's EOL now

Reply:
Yes, it is found that it only exists in Spark 2.4 + Hive 1.x, but there are still many people using 2.4 +, it is recommended to provide a patch solution

Reply:
OK, this is targeted at master though, and there will be no more 2.4 releases

Reply:
@SelfImpr001 Please upgrade your Spark version to the latest Spark version. 2.4.8 is the last release and no more 2.4.x releases should be expected even for bug fixes. Please see: https://spark.apache.org/versioning-policy.html

Reply:
@srowen @wangyum 
Thank you for your answer, because there are many developers whose environment is still in 2.4.x, and upgrading to the latest version requires a lot of verification. It is difficult to upgrade in a short time. In order to avoid the influence of the old version on the developer, it is recommended to merge the PR to the 2.4.X version, so that Doing this will not affect the master branch. If it cannot be merged due to version management specifications, just close this PR directly.

Reply:
We're saying the project no longer makes 2.4.x releases at all, so it wouldn't help anyone on 2.4

Reply:
Ok, thank you for your answers, I will close the PR now